File: Data Structures & Algorithms/palindromic-substrings/submission-1.py
class Solution:

    def palindromeString(self, s, left, right):
        count = 0
        n = len(s)

        while left >= 0 and right < len(s) and s[left] == s[right]:
            count += 1
            right += 1
            left -= 1

        return count


    def countSubstrings(self, s: str) -> int:
        n = len(s)
        count = 0
        for i in range(n):

            count += self.palindromeString(s, i, i)
            count +=  self.palindromeString(s, i , i + 1)

        return count

File: Data Structures & Algorithms/same-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} p
     * @param {TreeNode} q
     * @return {boolean}
     */
    isSameTree(p, q) {

        const queue = [];

        queue.push(p);
        queue.push(q);


        while(queue.length > 0)
        {
            const first = queue.shift();
            const second = queue.shift();


            if(first == null && second == null )
            {
                continue;
            }else{
                if(first == null || second == null || first.val != second.val)
                {
                    return false;
                }
            }


            queue.push(first.left);
            queue.push(second.left);


            queue.push(first.right);
            queue.push(second.right);
        }

        return true;

    }
}
